crates/red_knot_python_semantic/src/types.rs

```diff
@@ -463,6 +463,58 @@ impl<'db> Type<'db> {
         self == other
     }
 
+    /// Return true if there is just a single inhabitant for this type.
+    ///
+    /// Note: This function aims to have no false positives, but might return `false`
+    /// for more complicated types that are actually singletons.
+    pub(crate) fn is_singleton(self, db: &'db dyn Db) -> bool {
+        match self {
+            Type::Any
+            | Type::Never
+            | Type::Unknown
+            | Type::Todo
+            | Type::Unbound
+            | Type::Module(..)
```

Comment:
Oh! Modules are singletons.

Reply:
(Obviously `types.ModuleType` is not a singleton -- there are multiple different Python modules out there in the wild. But `Type::Module(<the typing module>)` is a singleton, as is `Type::Module(<the functools module>)`.)